src/main/java/org/opensearch/knn/plugin/KNNPlugin.java

```diff
@@ -342,12 +343,26 @@ public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings sett
     }
 
     /**
-     * Plugin can provide additional node settings, that includes new settings or overrides for existing one from core.
+     * Plugin can provide additional index settings, that includes new settings or overrides for existing one from core.
      *
      * @return settings that are set by plugin
      */
     @Override
-    public Settings additionalSettings() {
```

Comment:
Do we need to keep it for cluster level as well for indices that upgrade or were created on an old version of the cluster?

Reply:
I think it's not required as we apply the setting in the code, it's not persisted. Also if customer used custom index-level overwrites that will apply to 2.6 as well. Customer values should overwrite both cluster defaults and index provided by the plugin.

Reply:
Right, but assume that a customer did not apply any override. On 2.5, by default, their index with use vex/vec in the setting because we apply the cluster override. However, in 2.6, if no cluster override is provided, the index will not use vex/vec in the setting. So this could be an unexpected degradation.

Reply:
Checked with quick bwc test, indexes from 2.5 do not have index level setting when opened in 2.6, setting is stored along with the index and doesn't apply on index or shard opening as I was initially thinking.